dgp/core/models/project.py

```diff
@@ -0,0 +1,305 @@
+        elif _type == Path.__name__:
+            return Path(*params.values())
+        else:
+            # Handle project entity types
+            klass = {self._klass.__name__: self._klass, **project_entities}.get(_type, None)
+        if klass is None:  # pragma: no cover
+            raise AttributeError("Unhandled class %s in JSON data. Class is not defined"
+                                 " in entity map." % _type)
+        instance = klass(**params)
+        if parent is not None:
+            self._child_parent_map[instance.uid] = parent
+        self._registry[instance.uid] = instance
+        return instance
+
+
+class GravityProject:
+    def __init__(self, name: str, path: Union[Path], description: Optional[str] = None,
+                 create_date: Optional[datetime.datetime] = None,
+                 modify_date: Optional[datetime.datetime] = None,
+                 uid: Optional[str] = None, **kwargs):
+        self.uid = uid or OID(self, tag=name)
+        self.uid.set_pointer(self)
+        self._name = name
+        self._path = path
+        self._projectfile = PROJECT_FILE_NAME
+        self._description = description or ""
+        self.create_date = create_date or datetime.datetime.utcnow()
+        self.modify_date = modify_date or datetime.datetime.utcnow()
+
+        self._gravimeters = kwargs.get('gravimeters', [])  # type: List[Gravimeter]
+        self._attributes = kwargs.get('attributes', {})  # type: Dict[str, Any]
+
+    @property
+    def name(self) -> str:
+        return self._name
+
+    @name.setter
+    def name(self, value: str) -> None:
+        self._name = value.strip()
+        self._modify()
+
+    @property
+    def path(self) -> Path:
+        return Path(self._path)
+
+    @path.setter
+    def path(self, value: str) -> None:
+        self._path = Path(value)
+
+    @property
+    def description(self) -> str:
+        return self._description
+
+    @description.setter
+    def description(self, value: str):
+        self._description = value.strip()
+        self._modify()
+
+    @property
+    def gravimeters(self) -> List[Gravimeter]:
+        return self._gravimeters
+
+    def get_child(self, child_id: OID):
+        return [meter for meter in self._gravimeters if meter.uid == child_id][0]
+
+    def add_child(self, child) -> None:
+        if isinstance(child, Gravimeter):
+            self._gravimeters.append(child)
+            self._modify()
+        else:
+            raise TypeError("Invalid child type: {!r}".format(child))
+
+    def remove_child(self, child_id: OID) -> bool:
+        child = child_id.reference  # type: Gravimeter
+        if child in self._gravimeters:
+            self._gravimeters.remove(child)
+            return True
+        return False
+
+    def __repr__(self):
+        return '<%s: %s/%s>' % (self.__class__.__name__, self.name, str(self.path))
+
+    # TODO: Are these useful, or just fluff that should be removed
```

Comment:
Is the intention to protect the attributes initially defined by making a separate `_attributes` dictionary?

Unless there's another use case that I'm missing, then I think these can be safely removed.